sdks/go/sdk.go

```diff
@@ -76,3 +83,33 @@ func (s *SDK) Health() error {
 func (s *SDK) GameServer() (*sdk.GameServer, error) {
 	return s.client.GetGameServer(s.ctx, &sdk.Empty{})
 }
+
+// WatchGameServer asynchronously calls the given GameServerCallback with the current GameServer
+// configuration when the backing GameServer configuration is updated.
+// This function can be called multiple times to add more than one GameServerCallback.
+func (s *SDK) WatchGameServer(f GameServerCallback) error {
+	stream, err := s.client.WatchGameServer(s.ctx, &sdk.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "could not watch gameserver")
+	}
+
+	go func() {
```

Comment:
@markmandel Should this goroutine ever exit?

Reply:
Yes - on EOF of the gRPC stream. Basically when the stream closes on the sdk sidecar server shutdown.

I could have written the if statement much cleaner looking at it now.